plugins/adrv9009_adv.c

```diff
@@ -731,26 +737,31 @@ static const char *adrv9009_adv_sr_attribs[] = {
 	"debug.adrv9009-phy.adi,default-initial-calibrations-mask",
 };
 
-static void reload_settings(void)
+static void reload_settings()
 {
 	struct osc_plugin *plugin;
 	GSList *node;
 
 	for (node = plugin_list; node; node = g_slist_next(node)) {
 		plugin = node->data;
 
-		if (plugin && !strncmp(plugin->name, "adrv9009", 12)) {
+		if (plugin && !strncmp(plugin->name, "adrv9009", 12)) { // TO DO: How do we know the first plugin we find is not this one?
 			if (plugin->handle_external_request) {
 				g_usleep(1 * G_USEC_PER_SEC);
-				plugin->handle_external_request("Reload Settings");
+				plugin->handle_external_request(plugin, "Reload Settings");
 			}
 		}
 	}
 }
 
 static void signal_handler_cb(GtkWidget *widget, gpointer data)
 {
-	struct w_info *item = data;
+	struct data {
```

Comment:
why not define a single struct type ?

Reply:
I guess I didn't find a good name for this structure so I hid it inside the two functions where it was needed. I will declare this structure only once.